packages/grid/test/column-group.test.js

```diff
@@ -233,4 +163,140 @@ describe('column group', () => {
+      group.hidden = true;
+
+      const visibleColumn = document.createElement('vaadin-grid-column');
+      visibleColumn.hidden = false;
+      group.appendChild(visibleColumn);
+      group._observer.flush();
+
+      expect(visibleColumn.hidden).to.be.true;
+    });
+
+    it('should not propagate hidden to added column when group is visible', () => {
+      const hiddenColumn = document.createElement('vaadin-grid-column');
+      hiddenColumn.hidden = true;
+      group.appendChild(hiddenColumn);
+      group._observer.flush();
+
+      expect(hiddenColumn.hidden).to.be.true;
+    });
+
+    it('should not unhide when adding a visible column', () => {
+      group.hidden = true;
+
+      const visibleColumn = document.createElement('vaadin-grid-column');
+      visibleColumn.hidden = false;
+      group.appendChild(visibleColumn);
+      group._observer.flush();
+
+      expect(columns[1].hidden).to.be.true;
+    });
+
+    it('should not unhide when column is made visible', () => {
+      group.hidden = true;
+      columns[0].hidden = false;
+
+      expect(group.hidden).to.be.true;
+    });
+
+    it('should not unhide other columns when a column is made visible', () => {
+      group.hidden = true;
+      columns[0].hidden = false;
+
+      expect(columns[1].hidden).to.be.true;
+    });
+  });
+
+  describe('auto-hide', () => {
+    it('should auto-hide group when all columns are hidden', () => {
+      columns[0].hidden = true;
+      columns[1].hidden = true;
+
+      expect(group.hidden).to.be.true;
+      expect(group._autoHidden).to.be.true;
+    });
+
+    it('should auto-hide the group when removing all columns', () => {
+      group.removeChild(columns[0]);
+      group.removeChild(columns[1]);
+      group._observer.flush();
+
+      expect(group.hidden).to.be.true;
+      expect(group._autoHidden).to.be.true;
+    });
+
+    it('should unhide when column is made visible while auto-hidden', () => {
+      columns[0].hidden = true;
+      columns[1].hidden = true;
+      expect(group.hidden).to.be.true;
+      expect(group._autoHidden).to.be.true;
+
+      columns[0].hidden = false;
+      expect(group.hidden).to.be.false;
+      expect(group._autoHidden).to.be.false;
+    });
+
+    it('should unhide when adding a visible column while auto-hidden', () => {
+      group.removeChild(columns[0]);
+      group.removeChild(columns[1]);
+      group._observer.flush();
+
+      group.appendChild(columns[0]);
+      group._observer.flush();
+
+      expect(group.hidden).to.be.false;
+    });
+
+    it('should not unhide the group when adding a hidden column while auto-hidden', () => {
+      group.removeChild(columns[0]);
+      group.removeChild(columns[1]);
+      group._observer.flush();
+
+      columns[0].hidden = true;
+      group.appendChild(columns[0]);
+      group._observer.flush();
+
+      expect(group.hidden).to.be.true;
+    });
+
+    it('should not propagate hidden to added columns while auto-hidden', () => {
```

Comment:
This test should cover the actual bugfix. Note that propagating hidden includes both true and false, neither should be applied to child columns while auto-hidden.